File: backend/src/main/java/com/github/valfink/backend/fooditem/FoodItem.java
package com.github.valfink.backend.fooditem;

import com.github.valfink.backend.geolocation.Location;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document("foodItems")
public record FoodItem(
        @Id
        String id,
        String donatorId,
        String title,
        String photoUri,
        Location location,
        Instant pickupUntil,
        Instant consumeUntil,
        String description
) {
}

File: frontend/src/model/FoodItem.ts
import {User} from "./User";
import {Location} from "./Location";

export type FoodItem = {
    id?: string;
    title: string;
    photoUri: string;
    location: Location;
    pickupUntil: string;
    consumeUntil: string;
    description: string;
    donator: User;
}
